src/audio/pcm_converter/pcm_converter.c

```diff
@@ -28,8 +35,19 @@ void pcm_convert_as_linear(const struct audio_stream *source, uint32_t ioffset,
 	int chunk;
 	int N1, N2;
 
-	assert(audio_stream_get_avail_samples(source) >= samples + ioffset);
-	assert(audio_stream_get_free_samples(sink) >= samples + ooffset);
+	/* assert enough avail/free samples in source and sink buffer */
+	if (audio_stream_get_avail_samples(source) < samples + ioffset) {
+		tr_err(&pcm_conv_tr, "Available samples %d in source < samples to convert %d + offset %d",
+		       audio_stream_get_avail_samples(source), samples,
+		       ioffset);
+		return;
+	}
+	if (audio_stream_get_free_samples(sink) < samples + ooffset) {
+		tr_err(&pcm_conv_tr, "Free samples %d in sink < samples to convert %d + offset %d",
+		       audio_stream_get_avail_samples(source), samples,
```

Comment:
same comment, use a local variable - which would also have avoided the copy-paste mistake and using the wrong function.... Grrr.

Reply:
updated